File: life_cost_tracker/lib/domain/entities/expense_category.dart
// expense_category.dart
// LifeCostTracker
// Created by LifeCostTracker Team

enum ExpenseCategory {
  food,
  transport,
  shopping,
  entertainment,
  utilities,
  other;

  String get displayName {
    switch (this) {
      case ExpenseCategory.food:
        return '餐饮';
      case ExpenseCategory.transport:
        return '交通';
      case ExpenseCategory.shopping:
        return '购物';
      case ExpenseCategory.entertainment:
        return '娱乐';
      case ExpenseCategory.utilities:
        return '生活缴费';
      case ExpenseCategory.other:
        return '其他';
    }
  }

  String get systemIconName {
    switch (this) {
      case ExpenseCategory.food:
        return 'restaurant';
      case ExpenseCategory.transport:
        return 'directions_car';
      case ExpenseCategory.shopping:
        return 'shopping_bag';
      case ExpenseCategory.entertainment:
        return 'gamepad';
      case ExpenseCategory.utilities:
        return 'house';
      case ExpenseCategory.other:
        return 'more_horiz';
    }
  }
}

File: LifeCostTracker/Domain/Entities/ExpenseCategory.swift
//
//  ExpenseCategory.swift
//  LifeCostTracker
//
//  Created by LifeCostTracker Team
//

import Foundation

enum ExpenseCategory: String, CaseIterable, Codable {
    case food
    case transport
    case shopping
    case entertainment
    case utilities
    case other
    
    var displayName: String {
        switch self {
        case .food:
            return "餐饮"
        case .transport:
            return "交通"
        case .shopping:
            return "购物"
        case .entertainment:
            return "娱乐"
        case .utilities:
            return "生活缴费"
        case .other:
            return "其他"
        }
    }
    
    var systemImageName: String {
        switch self {
        case .food:
            return "fork.knife"
        case .transport:
            return "car"
        case .shopping:
            return "bag"
        case .entertainment:
            return "gamecontroller"
        case .utilities:
            return "house"
        case .other:
            return "ellipsis.circle"
        }
    }
}
